src/pages/Account/Venue/VenueMapEditor/Container.tsx

```diff
@@ -0,0 +1,242 @@
+            box.left,
+            "width",
+            safeZonePerc.width
+          ),
+        } as SubVenueIcon)
+    );
+
+    onChange && onChange(unscaledBoxes);
+  }, [boxes, onChange, imageDims, safeZonePerc.width, safeZonePerc.height]);
+
+  useMemo(() => {
+    if (!imageDims || !imageDims.width || !imageDims.height) return;
+
+    const copy = iconsMapPerc.map(
+      (mapIcon) =>
+        ({
+          ...mapIcon,
+          width: imageDims.width * mapIcon.width * safeZonePerc.width,
+          height: imageDims.height * mapIcon.height * safeZonePerc.height,
+          top:
+            imageDims.height * ((1 - safeZonePerc.height) / 2) +
+            imageDims.height * mapIcon.top * safeZonePerc.height,
+          left:
+            imageDims.width * ((1 - safeZonePerc.width) / 2) +
+            imageDims.width * mapIcon.left * safeZonePerc.width,
+        } as SubVenueIcon)
+    );
+
+    setBoxes(copy);
+  }, [iconsMapPerc, imageDims, safeZonePerc]);
+
+  const moveBox = useCallback(
+    (id: number, left: number, top: number) => {
+      setBoxes(
+        update(boxes, {
+          [id]: {
+            $merge: { left, top },
+          },
+        })
+      );
+    },
+    [boxes]
+  );
+  const resizeBox = useCallback(
+    (id: number) => (dimensions: Dimensions) => {
+      const { width, height } = dimensions;
+      setBoxes(
+        update(boxes, {
+          [id]: {
+            $merge: { width, height },
+          },
+        })
+      );
+    },
+    [boxes]
+  );
+
+  const [, drop] = useDrop({
+    accept: ItemTypes.SUBVENUE_ICON,
+    drop: (item: DragItem, monitor) => {
+      if (!interactive) return;
+      const delta = monitor.getDifferenceFromInitialOffset() as {
+        x: number;
+        y: number;
+      };
+
+      let left = Math.round(item.left + delta.x);
+      let top = Math.round(item.top + delta.y);
+      if (snapToGrid) {
+        [left, top] = doSnapToGrid(left, top);
+      }
+
+      moveBox(item.itemId, left, top);
+    },
+  });
+
+  return (
+    <>
+      <div ref={drop}>
+        <div ref={ref} className={styles.mapContainer}>
+          <ReactResizeDetector handleWidth handleHeight>
+            {({ targetRef }) => <span ref={targetRef} />}
+          </ReactResizeDetector>
+          <img
+            alt="draggable background"
+            style={{
+              width: "100%",
+            }}
+            src={backgroundImage}
+          />
+          {boxes.map((box, idx) => (
+            <DraggableSubvenue
+              key={idx}
+              itemId={idx}
+              rounded={!!rounded}
+              {...box}
+              onChangeSize={resizeBox(idx)}
+              lockAspectRatio={lockAspectRatio}
+              onDragStart={setDragItemId}
+              isSaving={isSaving}
+            />
+          ))}
+        </div>
+      </div>
+      {imageDims && interactive && (
+        <CustomDragLayer
+          snapToGrid={!!snapToGrid}
+          rounded={!!rounded}
```

Comment:
Not sure where these come from, but seeing it `!!rounded` used several times, it might be more clear to have something like

```tsx
const isRounded: boolean = !!rounded;
```

before the `return` keyword.